mlir/test/Dialect/Vector/vector-transfer-flatten.mlir

```diff
@@ -459,6 +459,26 @@ func.func @fold_unit_dims_entirely(%arg0 : vector<8xi32>,
 // CHECK-128B-LABEL: func @fold_unit_dims_entirely(
 //   CHECK-128B-NOT:   memref.collapse_shape
 
+// -----
+
+func.func @fold_unit_center_dim_scalable(%arg0 : vector<8x1x[1]xf128>,
```

Comment:
This should work for both scalable and fixed-width vectors, right? Why not add test for both? Also, nit, “central” (which means exactly in the middle) —> “inner” (neither leading nor trailing).

Reply:
Well unit scalable dims should not be eliminated. but it is supported. Added a test.